include/kernel/sporadic.h

```diff
@@ -55,29 +55,49 @@ static inline word_t refill_absolute_max(cap_t sc_cap)
 /* @return the current amount of empty slots in the refill buffer */
 static inline word_t refill_size(sched_context_t *sc)
 {
-    if (sc->scRefillHead <= sc->scRefillTail) {
-        return (sc->scRefillTail - sc->scRefillHead + 1u);
-    }
-    return sc->scRefillTail + 1u + (sc->scRefillMax - sc->scRefillHead);
+    return sc->scRefillCount;
 }
 
 /* @return true if the circular buffer of refills is current full (all slots in the
  * buffer are currently being used */
 static inline bool_t refill_full(sched_context_t *sc)
 {
-    return refill_size(sc) == sc->scRefillMax;
+    return sc->scRefillCount == sc->scRefillMax;
+}
+
+static inline bool_t refill_empty(sched_context_t *sc)
```

Comment:
I don't understand what `refill_empty` signifies? Before this commit, the smallest size of the queue was when `sc->scRefillHead == sc->scRefillTail` signifying that the queue had a single element (assuming `scRefillMax > 0`).
Now it seems that `scRefillCount == 1` is the new way of checking this.  When would `refill_empty` return true and so how has `schedule_used` been changed to incorporate it (it removes `refill_single` and adds `refill_empty`)?

Reply:
`refill_split_check` changed to pop the head off of the refill queue when charging the head wouldn't leave a large enough refill to be larger than `MIN_BUDGET`. In this case, when the charged refill is added back to the end, the queue may now be empty.

Reply:
Right, but in this commit when the queue has a single refill, `refill_split_check` is updating it in place and not popping off the head.  So there isn't a way for refill_empty to return true.